packages/app/cypress/e2e/integration/runs.spec.ts

```diff
@@ -1,80 +1,256 @@
-describe('App: Runs Page', () => {
+import type { Interception } from '@packages/net-stubbing/lib/external-types'
+
+describe('App: Runs', () => {
   beforeEach(() => {
     cy.scaffoldProject('component-tests')
     cy.openProject('component-tests')
     cy.startAppServer()
   })
 
-  it('resolves the runs page', () => {
-    cy.loginUser()
-    cy.visitApp()
-    cy.wait(1000)
-    cy.get('[href="#/runs"]').click()
-    cy.get('[data-cy="runs"]')
+  context('Runs Page', () => {
+    it('resolves the runs page', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.get('[href="#/runs"]', { timeout: 1000 }).click()
+      cy.get('[data-cy="runs"]')
+      cy.get('[data-cy="app-header-bar"]').findByText('Runs').should('be.visible')
+    })
+
+    it('shows the loader', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        await new Promise((resolve) => setTimeout(resolve, 200))
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs-loader"]')
+      cy.get('[data-cy="runs"]')
+    })
   })
 
-  it('shows the loader', () => {
-    cy.loginUser()
-    cy.remoteGraphQLIntercept(async (obj) => {
-      await new Promise((resolve) => setTimeout(resolve, 200))
+  context('Runs - Login', () => {
+    it('when logged out, shows call to action', () => {
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Log In').should('exist')
```

Comment:
Should "design" or "marketing" choose to replace "Log In" by "Sign in" or even "Log in", this tests would fail even if it should not.

You can use the `defaultMessages` instead of hardcoded texts, your tests will be more resilient. An additional bonus: no more typos and you get autocompletion.

Reply:
Additionally, this is not the call to action I would have expected. 
It looks at the Top Nav instead of looking at the runs page who should tell the user what to do.

Reply:
```suggestion
      cy.contains(defaultMessages.runs.connect.buttonUser).should('exist')
```